projects/plugins/jetpack/_inc/lib/core-api/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-blog-stats.php

```diff
@@ -0,0 +1,81 @@
+<?php
+/**
+ * Get blog stats.
+ *
+ * @package automattic/jetpack
+ */
+
+use Automattic\Jetpack\Stats\WPCOM_Stats;
+
+/**
+ * Blog Stats block endpoint.
+ */
+class WPCOM_REST_API_V2_Endpoint_Blog_Stats extends WP_REST_Controller {
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+		add_action( 'rest_api_init', array( $this, 'register_routes' ) );
+	}
+
+	/**
+	 * Register endpoint routes.
+	 */
+	public function register_routes() {
+		register_rest_route(
+			'wpcom/v2',
+			'/blog-stats',
+			array(
+				array(
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_blog_stats' ),
+					'permission_callback' => function () {
+						return current_user_can( 'edit_posts' );
+					},
+					'args'                => array(
+						'post_id' => array(
+							'description'       => __( 'Post ID to obtain stats for.', 'jetpack' ),
+							'type'              => array( 'string', 'integer' ),
+							'required'          => false,
+							'validate_callback' => function ( $param ) {
+								return is_numeric( $param );
+							},
+						),
+					),
+				),
+			)
+		);
+	}
+
+	/**
+	 * Get the blog stats.
+	 *
+	 * @param \WP_REST_Request $request Request object.
+	 *
+	 * @return array Blog stats.
+	 */
+	public function get_blog_stats( $request ) {
+		$post_id   = $request->get_param( 'post_id' );
+		$post_data = convert_stats_array_to_object(
+			( new WPCOM_Stats() )->get_post_views( $post_id, array( 'fields' => 'views' ) )
```

Comment:
The one problem we'll face with this implementation is that this will blow up folks' options table when they use the block in a page template. 

Every time we fetch post views for a post, we cache the data in a transient. If we start regularly fetching views for all posts on a site, this will make for a lot of transients. I'm not sure we'd want that to happen.

cc @Automattic/red for their opinion on this.

Reply:
I'm afraid this would be problematic on sites with a lot of posts. 

I'll let folks from @Automattic/red chime in on this. Unfortunately this is a tricky problem.

Reply:
Thanks for working on this! It looks amazing @Aurorum 

Thanks for your feedback @jeherve 

This is a tricky indeed. Additionally, the block could possibly have impact on user's page performance, as the data needs to be refreshed every 5 mins. So we really want to be really cautious about it.

Could we cache in post meta table? Would that potentially blow it up as well?

Another thought is we could go totally async calls to API without any cache.

looping in more crew friends for advice @Automattic/jetpack-crew 😄

Reply:
Thanks so much @kangzj! Those both sound like viable alternatives to me. 

Let me know if anyone has a preference, and I’ll try to include it here.

Reply:
> we could go totally async calls to API without any cache - frontend loads the views from an API which doesn't cache data on the site.

👍 This may be the lightest solution, it wouldn't clog databases and shouldn't have as big an impact on the frontend.

Reply:
Thanks! I've added that now. :)

Reply:
> 👍 This may be the lightest solution, it wouldn't clog databases and shouldn't have as big an impact on the frontend.

As long as it doesn't result in every frontend page view hitting the API or database. If that happens, it'll probably wind up tripping some WordPress.com monitor and we'll have to add some sort of caching back in.

Reply:
Hm, good point.

Reply:
If it’s more trouble than it’s worth, I can always just remove the feature - seems a popular request (and a bit defeatist!!) though.

Is there a downside to caching in the post meta table?

Reply:
I've gone with @kangzj's idea of caching in the post meta since it actually looks like that's what the Related Posts feature already does, and it seems to resolve the problem here. 

Let me know if I should change it!